common/arg.cpp

```diff
@@ -298,6 +272,9 @@ static bool common_download_file_single(const std::string & url, const std::stri
         // if we cannot open the metadata file, we assume that the downloaded file is not valid (etag and last-modified are left empty, so we will download it again)
     } else {
         LOG_INF("%s: no previous model file found %s\n", __func__, path.c_str());
+        if (offline) {
+            return false;
```

Comment:
This code branch will be trigger when `file_exists == false && offline == true`, so technically one of the condition inside `if (offline)` below now become dead code.

Instead, you can move the `if (offline)` into `if (file_exists)` above. So if file exist, we completely skip reading metadata

Reply:
ah ofc, rearranged to skip, thanks!